data_collection/web/guides.rst

```diff
@@ -10,3 +10,35 @@ Downloads and Outlinks
 ----------------------
 
 Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop p
+
+Site search
+-----------
+
+Our data collection engine automatically converts page views into site search events if the URL contains site-search query parameters. You can use :ref:`trackSiteSearch<js-api-trackSiteSearch>` method to provide additional site-search-related information (like results count) or track uses of reload-free search boxes.
+
+:ref:`trackSiteSearch<js-api-trackSiteSearch>` accepts the following parameters:
+
+* **keyword** - what term someone looked for
+* **category** (optional) - which category the search was in
+* **results** (optional) - how many search results were returned
+* **dimensions** (optional) - additional event-scoped custom dimensions to send along the site search
+
+It is used like this:
+
+.. code:: javascript
+
+    _paq.push(["trackSiteSearch", "les paul", "electric guitars", 5, { dimension10: "amber" }]);
+
+In this case, we track site search with keyword *les paul*, category *electric guitars*, *5* search results and event-scoped custom dimension *10* with value *amber*.
+
+The optional parameters might be skipped or replaced with ``false`` to indicate no value.
+
+.. code:: javascript
+
+    _paq.push(["trackSiteSearch", "playstation"]); // only keyword provided
+    _paq.push(["trackSiteSearch", "playstation", "consoles"]); // only keyword and category provided
+    _paq.push(["trackSiteSearch", "playstation", false, 5]); // only keyword and results count provided
+
+.. warning::
+
+    Calls to ``trackSiteSearch`` should replace the usual ``trackPageView`` if submitting a search form reloads the page.
```

Comment:
I have a hard time with understanding logic here.

This replacement of `trackPageView` with `trackSiteSearch` does make sense if user wants to reproduce behavior of automated detection of search query in URL by Piwik PRO. Maybe this is warning to avoid double count for searches when user is calling `trackSiteSearch` on forms that send their data via GET requests? In that case if user will use `trackPageView` and `trackSiteSearch` on results page, then we'll count it as 2 separate searches.

BTW: Page reload isn't the deciding factor when this function is useful (e.g. search send via POST).

Reply:
See if the new version is clearer.